### PR TITLE
Add fibonacci circuit example

### PR DESCRIPTION
**Pull request by dysfunctor**
Played around with rcc and made a fibonacci circuit that could be another example

### CODE REVIEW

File: circuit-examples/examples/halo2_fibonacci.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    assert!(which_fib_number > 3);
+
+    let first_fib_number = input_wire("first_fib_number");
+    let second_fib_number = input_wire("second_fib_number");
+    let output_fib_number = input_wire("output_fib_number");
+
+    let mut fib_numbers = Vec::new();
+
+    fib_numbers.push(first_fib_number);
+    fib_numbers.push(second_fib_number);
```
Comment (clvv) [inline circuit-examples/examples/halo2_fibonacci.rs]: Let's use the `vec` macro here?
```
let mut fib_numbers = vec![first_fib_number, first_fib_number]
```

File: circuit-examples/examples/halo2_fibonacci.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    first_fib_number.declare_public("first_fib_number");
+    second_fib_number.declare_public("second_fib_number");
+
+    let last_fib_number = *fib_numbers.last().unwrap();
+    last_fib_number.declare_public("last_fib_number");
+
+    // Constrains the last fibonacci number to equal the third public input.
+    //
+    assert!(last_fib_number == output_fib_number);
```
Comment (clvv) [inline circuit-examples/examples/halo2_fibonacci.rs]: The `assert!` here should be removed. `==` operator on `Wire`s is equivalent to `last_fib_number.assert_eq(output_fib_number)` and always return true, by convention.

**Comment by dysfunctor** [comment]
Ok, addressed comments

**Comment by dysfunctor** [comment]
Removed the extra input wire in [92cc539](https://github.com/clvv/rcc/pull/2/commits/92cc53985e0355c22e12251e98e45a8b0fdc9eeb)

**Comment by clvv** [comment]
Merging! Thanks for coding up this example.

**Pull request merged**